src/main/scala/components/memory/memory-combin-ports.scala

```diff
@@ -0,0 +1,131 @@
+// Combinational/'synchronous' memory module
+
+package dinocpu
+
+import chisel3._
+import chisel3.util._
+import dinocpu.MemoryOperation._
+
+/**
+  * The instruction memory port.
+  *
+  * The I/O for this module is defined in [[IMemPortIO]].
+  */
+class ICombinMemPort extends BaseIMemPort {
+  // When the pipeline is supplying a high valid signal
+  when (io.valid) {
+    val request = Wire(new Request)
+    request := DontCare
+    request.address   := io.address
+    request.operation := Read
+
+    io.request.bits  := request
+    io.request.valid := true.B
+  } .otherwise {
+    io.request.valid := false.B
+  }
+
+  // When the memory is outputting a valid instruction
+  io.good := io.response.valid
+  when (io.response.valid) {
```

Comment:
This when is probably not necessary. The pipeline should ignore the `io.instruction` if `!io.good`